packages/thirdweb/src/utils/contract/convert-param-values.ts

```diff
@@ -0,0 +1,84 @@
+import { isAddress } from "../address.js";
+import { isHex, padHex } from "../encoding/hex.js";
+
+/**
+ * Converts an array of parameter values to their respective types based on the provided type array.
+ *
+ * This utility function is particularly useful for ensuring that parameter values are correctly formatted
+ * according to the expected types before they are used in further processing or passed to a Solidity smart contract.
+ *
+ * @param {string[]} constructorParamTypes - An array of type strings indicating the expected types of the values,
+ *                   following Solidity type conventions (e.g., "address", "uint256", "bool").
+ * @param {any[]} constructorParamValues - An array of values to be converted according to the types.
+ * @returns {any[]} - An array of values converted to their respective types.
+ *
+ * @example
+ * ```ts
+ * import { convertParamValues } from "thirdweb/utils";
+ *
+ * const example1 = convertParamValues(
+ *   ["address", "uint256"],
+ *   ["0x.....", "1200000"]
+ * ); // result: ["0x......", 1200000n]
+ *
+ * const example2 = convertParamValues(
+ *   ["address", "bool"],
+ *   ["0x.....", "true"]
+ * ); // result: ["0x......", true]
+ * ```
+ */
+export function convertParamValues(
```

Comment:
i think `parseAbiParams` i a better name for this.

also remember to tag this TSDoc as `@utils`